libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_subscription.c

```diff
@@ -370,97 +457,33 @@ TEST( MQTT_Unit_Subscription, ListInsertRemove )
 TEST( MQTT_Unit_Subscription, ListFindByTopicFilter )
 {
     _mqttSubscription_t * pSubscription = NULL;
-    IotLink_t * pSubscriptionLink = NULL;
     _topicMatchParams_t topicMatchParams = { 0 };
+    int8_t index = -1;
 
     topicMatchParams.pTopicName = "/test0";
     topicMatchParams.topicNameLength = 6;
 
     /* On empty list. */
-    pSubscriptionLink = IotListDouble_FindFirstMatch( &( _pMqttConnection->subscriptionList ),
-                                                      NULL,
-                                                      IotTestMqtt_topicMatch,
-                                                      &topicMatchParams );
-    TEST_ASSERT_EQUAL_PTR( NULL, pSubscriptionLink );
+    index = IotMqtt_FindFirstMatch( &( connToContext[ contextIndex ].subscriptionArray ),
+                                    0,
+                                    &topicMatchParams );
+    TEST_ASSERT_EQUAL( -1, index );
 
     _populateList();
 
     /* Topic filter present. */
-    pSubscriptionLink = IotListDouble_FindFirstMatch( &( _pMqttConnection->subscriptionList ),
-                                                      NULL,
-                                                      IotTestMqtt_topicMatch,
-                                                      &topicMatchParams );
-    TEST_ASSERT_NOT_EQUAL( NULL, pSubscriptionLink );
-    pSubscription = IotLink_Container( _mqttSubscription_t, pSubscriptionLink, link );
-    TEST_ASSERT_NOT_EQUAL( NULL, pSubscription );
+    index = IotMqtt_FindFirstMatch( &( connToContext[ contextIndex ].subscriptionArray ),
+                                    0,
+                                    &topicMatchParams );
+    TEST_ASSERT_EQUAL( 0, index );
 
     /* Topic filter not present. */
     topicMatchParams.pTopicName = "/notpresent";
     topicMatchParams.topicNameLength = 11;
-    pSubscriptionLink = IotListDouble_FindFirstMatch( &( _pMqttConnection->subscriptionList ),
-                                                      NULL,
-                                                      IotTestMqtt_topicMatch,
-                                                      &topicMatchParams );
-    TEST_ASSERT_EQUAL_PTR( NULL, pSubscriptionLink );
-}
-
-/*-----------------------------------------------------------*/
-
-/**
- * @brief Tests searching the subscription list using a packet identifier.
- */
```

Comment:
Don't we have to create an equivalent test ?

Reply:
Addressed